code/Weapons/Weapon.cs

```diff
@@ -59,7 +61,19 @@ public override void AttackPrimary()
 			Fire();
 		}
 
-		public virtual void Fire() { }
+		public async virtual void Fire()
+		{
+			if ( !IsFiredTurnEnding )
+				return;
+
+			/* 
+			 * TODO: Let physics resolve and weapon to finish firing before ending the players turn.
+			 * Temporary delay to simulate this.
+			 */
+			await GameTask.DelaySeconds( 1 );
+
+			(Game.StateHandler?.State as PlayingState)?.Turn?.SetTimeRemaining( GameConfig.TurnTimeRemainingAfterFired );
```

Comment:
Not a huge fan of this either, could do with a `CurrentTurn` property or something. Also should `Turn` ever actually be null I think that should be checked before we actually fire the weapon